lib/connection/connection_config.js

```diff
@@ -208,49 +208,10 @@ function ConnectionConfig(options, validateCredentials, qaMode, clientInfo) {
   const proxyPassword = options.proxyPassword;
   const proxyProtocol = options.proxyProtocol || 'http';
   const noProxy = options.noProxy;
-
   // if we're running in node and some proxy information is specified
   let proxy;
-  if (Util.isNode() && (Util.exists(proxyHost) || Util.exists(proxyPort))) {
-    // check for missing proxyHost
-    Errors.checkArgumentExists(Util.exists(proxyHost),
-      ErrorCodes.ERR_CONN_CREATE_MISSING_PROXY_HOST);
-
-    // check for invalid proxyHost
-    Errors.checkArgumentValid(Util.isString(proxyHost),
-      ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_HOST);
-
-    // check for missing proxyPort
-    Errors.checkArgumentExists(Util.exists(proxyPort),
-      ErrorCodes.ERR_CONN_CREATE_MISSING_PROXY_PORT);
-
-    // check for invalid proxyPort
-    Errors.checkArgumentValid(Util.isNumber(proxyPort),
-      ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_PORT);
-
-    if (Util.exists(noProxy)) {
-      // check for invalid noProxy
-      Errors.checkArgumentValid(Util.isString(noProxy),
-        ErrorCodes.ERR_CONN_CREATE_INVALID_NO_PROXY);
-    }
-
-    if (Util.exists(proxyUser) || Util.exists(proxyPassword)) {
-      // check for missing proxyUser
-      Errors.checkArgumentExists(Util.exists(proxyUser),
-        ErrorCodes.ERR_CONN_CREATE_MISSING_PROXY_USER);
-
-      // check for invalid proxyUser
-      Errors.checkArgumentValid(Util.isString(proxyUser),
-        ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_USER);
-
-      // check for missing proxyPassword
-      Errors.checkArgumentExists(Util.exists(proxyPassword),
-        ErrorCodes.ERR_CONN_CREATE_MISSING_PROXY_PASS);
-
-      // check for invalid proxyPassword
-      Errors.checkArgumentValid(Util.isString(proxyPassword),
-        ErrorCodes.ERR_CONN_CREATE_INVALID_PROXY_PASS);
-
+  if (Util.isNode()) {
+    if ((Util.exists(proxyHost) || Util.exists(proxyPort))) {
```

Comment:
in this path, we try to set `user` and `password` for the proxy as well. but that's only for authenticated proxy. not all proxies need authentication. what happens if the proxy is unauthenticated and only `host` and `port` is set ?

Reply:
So, when the driver validates each options, the user and the password will be deleted if they are empty.